pydrad/parse/parse.py

```diff
@@ -525,6 +436,102 @@ def grid_edges_right(self) -> u.cm:
         """
         return self.grid_edges[1:]
 
+    @property
+    @u.quantity_input
+    def coordinate(self) -> u.cm:
+        """
+        Coordinate in the field-aligned direction
+        """
+        return self.grid_centers
+
+    @property
+    @u.quantity_input
+    def electron_mass_density(self) -> u.Unit('g cm-3'):
+        # TODO: Account for possible presence of both electron
+        # and ion mass densities. If the electron mass density
+        # is present in this file, it will shift all of the
+        # indices in the .amr file by one.
+        if 'electron_mass_density' in self._amr_data.colnames:
+            return self._amr_data['electron_mass_density']
+        return self.ion_mass_density
+
+    @property
+    @u.quantity_input
+    def ion_mass_density(self) -> u.Unit('g cm-3'):
+        return self._amr_data['ion_mass_density']
+
+    @property
+    @u.quantity_input
+    def momentum_density(self) -> u.Unit('g cm-2 s-1'):
+        return self._amr_data['momentum_density']
+
+    @property
+    @u.quantity_input
+    def electron_energy_density(self) -> u.Unit('erg cm-3'):
+        return self._amr_data['electron_energy_density']
+
+    @property
+    @u.quantity_input
+    def ion_energy_density(self) -> u.Unit('erg cm-3'):
+        return self._amr_data['ion_energy_density']
+
+    @property
+    @u.quantity_input
+    def velocity(self) -> u.cm/u.s:
+        if hasattr(self, '_phy_data'):
+            return self._phy_data['velocity']
+        return self.momentum_density / self.ion_mass_density
+
+    @property
+    @u.quantity_input
+    def ion_density(self) -> u.Unit('cm-3'):
+        if hasattr(self, '_phy_data'):
+            return self._phy_data['ion_density']
+        # NOTE: This is pulled directly from the HYDRAD source code
+        # https://github.com/rice-solar-physics/HYDRAD/blob/master/Resources/source/constants.h
+        # and is the average ion mass for a H-He plasma computed by weighting the H and He ion
+        # masses with a particular set of abundances.
+        m_ion = 2.171e-24 * u.g
```

Comment:
Would it be better or is it possible to pull this value directly from the HYDRAD package?  If the precision or abundance set is ever changed, for example?

Reply:
Yeah I found this unsatisfying as well. Pulling it from HYDRAD would be a pain and would mean parsing a `.h` file. Depending on HYDRAD itself would also be awkward since it isn't really a package per se. Beyond parsing it out of a results file itself, I'm not sure there's a great way to ensure the value is always consistent.

Similar to my comment about `gamma`, I would suggest we create our own constants file and import from there so that way at least this isn't as buried as it is now. I suspect this value isn't going to change much on the HYDRAD side.

Reply:
One other option is to save the multiplicative factor (1.whatever) that accounts for the other elements, and set this to 
```suggestion
        m_ion = mass_factor * const.m_p
```

It still has the issue that something needs to be hard-coded, though.

Reply:
I've now created a constants file from which this value is pulled with a reference to the file in HYDRAD.